### ISSUE TITLE
Unexpected ref safety error reported with `out _` with C#7.2 ref safety rules

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#65651]
Compile with .NET 6 and `-langversion:10`:
```csharp
using System;

class Program
{
    static Span<int> Test1()
    {
        var s = ReturnsSpan(out _);
        return s; // error CS8352: Cannot use variable 's' in this context ...
    }

    static Span<int> Test2()
    {
        var s = ReturnsSpan(out var _);
        return s;
    }

    static Span<int> Test3()
    {
        var s = ReturnsSpan(out var unused);
        return s;
    }

    static ref Span<int> ReturnsSpan(out Span<int> x)
    {
        x = default;
        return ref x;
    }
}
```

**Expected Behavior**:
No errors

**Actual Behavior**:
(8,16,8,17): error CS8352: Cannot use variable 's' in this context because it may expose referenced variables outside of their declaration scope

**Comment by RikkiGibson** [comment]
Why are no errors expected? If we compile under 7.2 then we expect the out argument may be returned by reference. I would just expect a similar error in Test1, Test2 and Test3. Right?

**Comment by cston** [comment]
The `Span<int>` instances are returned by value from `Test1`, `Test2`, `Test3`.
